libr/io/cache.c

```diff
@@ -159,5 +159,5 @@ R_API int r_io_cache_read(RIO *io, ut64 addr, ut8 *buf, int len) {
 			covered += l;
 		}
 	}
-	return covered;
+	return (covered == 0) ? false: true;
```

Comment:
I am with bool api. Normally the code does something like  `if (r_io_read_at (io, addr, buf, len) < len) break;` so at the end you are only interested whether the read was successful.